### ISSUE TITLE
[BUG] DF.app responding to SF.app conversations again.... did we lose a fix?

### ISSUE THREAD

**Issue by ear-dev** [WideChat/Rocket.Chat#1122]
https://chatbot.vega.viasat.com/live/SGBNxhziYWQpW3F7h/room-info

We are occasionally seeing this error again:
```
\"message\":\"Error occurred while processing handover. Details Error: The selected department and the current room department are the same [error-forwarding-chat-same-department]\"}"}
```
We had already fixed this, but did we maybe lose the fix somewhere?

**Comment by ear-dev** [comment]
![image](https://user-images.githubusercontent.com/28864717/159771540-52275f0d-2e3e-456c-8047-86e3f69c0faa.png)

**Comment by ear-dev** [comment]
![image](https://user-images.githubusercontent.com/28864717/159771686-273d9746-c271-456c-8269-4281ad563e37.png)

**Comment by ear-dev** [comment]
@AlexanderKanakis I think you worked on the original fix for this issue.....  can you take a look please when you have a chance to see if we may have lost the fix........  or could we have hit an edge case that we didn't cover?

**Comment by AlexanderKanakis** [comment]
Related PRs: 
- ~~https://github.com/WideChat/Apps.Salesforce.LiveAgent/pull/55~~
- ~~https://github.com/WideChat/Apps.Dialogflow/pull/150~~

Attempts to fix DF messages after escalation by checking for `handoverForEscalation` property in the room's persistent data, set to `true` by Apps.Salesforce.LiveAgent when initializing escalation. Set to `false` when closing chat, or when performing an additional handover from Apps.Salesforce.LiveAgent.

**Comment by ear-dev** [comment]
@Shailesh351 please verify this new approach to fixing this problem, and review the PRs when available.  If this is the better and more global approach, should we revert the previous attempts to fix?

Please chime in on this story to suggest how to proceed. 

- did we miss edge cases?
- do we need a bit more exploration?

**Comment by Shailesh351** [comment]
@AlexanderKanakis @ear-dev,

The issue we fixed earlier was different than this. It was regarding closing the room which was already closed. And I think it's working fine currently.

The error `Error occurred while processing handover. Details Error: The selected department and the current room department are the same [error-forwarding-chat-same-department]`. This occurs when DF App is trying to hand over the already handed over chat. From the logs, I found more sessions with this behavior.
https://chatbot.vega.viasat.com/live/nHYPKMBN8PGwxFqLy
https://chatbot.vega.viasat.com/live/ZWsCXw4pBvMWWBmRN
https://chatbot.vega.viasat.com/live/rdtfyTMd5aF4nrnvP
https://chatbot.vega.viasat.com/live/4jhvpBCWrRAWz2FZj

In these cases, I see multiple `Chat with agent` messages from the user or `An agent can assist with this.` from DF. So I think the user is clicking multiple times on the `Chat with agent button`.

<img width="880" alt="Screenshot 2022-04-06 at 2 09 15 PM" src="https://user-images.githubusercontent.com/18264684/161934536-1b40fa15-2b2c-44fb-86f6-de13eb17a746.png">
<img width="880" alt="Screenshot 2022-04-06 at 2 11 16 PM" src="https://user-images.githubusercontent.com/18264684/161934580-003efff1-9372-4071-b772-754799985135.png">
<img width="880" alt="Screenshot 2022-04-06 at 2 11 22 PM" src="https://user-images.githubusercontent.com/18264684/161934585-fb34d29e-f80a-4464-8d54-969242f26d8f.png">

@AlexanderKanakis Your PRs for adding `handoverForEscalation` are good to have to avoid messages from the DF app in the escalated chat. But I think we already have the `isHandedOverFromDialogFlow` flag in the room custom fields so we don't need to create a new flag and we can reuse the old one. Can you please check if that works? Thanks for the PRs 😊

**Comment by ear-dev** [comment]
Thanks for looking at this @Shailesh351 
Can you summarize for me please?  We would still like both of these PRs, or just one of them?  Or are you suggesting that one of them should possibly be adjusted?

Also, shouldn't this setting in the agent config.json prevent a visitor from clicking the 'chat with an agent' button twice?  `"hide_quickreplies": true,`

Thanks!

**Comment by Shailesh351** [comment]
@ear-dev Yes, we will have two PRs,

1) Which @AlexanderKanakis already created but requires modification
2) To debug and fix the behavior where we see multiple `Chat with agent` msgs.

>Also, shouldn't this setting in the agent config.json prevent a visitor from clicking the 'chat with an agent' button twice? "hide_quickreplies": true,

Yes, it should. But from the above chats looks there is some loophole and we need to debug that

**Comment by ear-dev** [comment]
@Shailesh351 please show where in the code the isHandedOverFromDialogFlow is being set.

**Comment by AlexanderKanakis** [comment]
Related PRs:
https://github.com/WideChat/Apps.Dialogflow/pull/153

**Comment by Shailesh351** [comment]
NOTE: The newly created story for the behavior where we see multiple Chat with agent msgs. issue https://github.com/WideChat/Rocket.Chat/issues/1137
